src/lightning/data/cache/worker.py

```diff
@@ -0,0 +1,345 @@
+# Copyright The Lightning AI team.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Copy pasted from https://github.com/pytorch/pytorch/blob/main/torch/utils/data/_utils/worker.py + lines 332-342
+# TODO: Delete me when this is addressed https://github.com/pytorch/pytorch/issues/110156
```

Comment:
Seems weird to have the copyright notice and the file is copied from elsewhere

Reply:
removed.